README.md

```diff
@@ -35,33 +35,139 @@ Here you can store a global array of configurations. Any project can read from t
+
+* `Compile and Link`: Compiles and links PSL or tables
+* `Get Element from Host`: Gets a new element
+* `Refresh from Host`: Refreshes an existing element
+* `Run PSL`: Runs a PSL element "as is" and displays the output
+* `Send to Host`: Sends an existing element
+* `Table Get from Host`: Gets all elements related to a table
+* `Table Refresh from Host`: Fetches all elements related to an existing table element
+* `Table Send to Host`: Sends all existing elements related to the table
+* `Test Compile`: Test compiles PSL
+
+> Please note that the Host Communication is done *asynchronously*, meaning that vscode will not require you to wait to finish one action before you start another. This may have unintended consequences if you do not wait. For example, you must wait for sending to finish before you compile and link.
+
+### Getting New Elements
+
+Two commands `Get Element from Host` and `Table Get from Host` will allow you to get new elements from the Host. When activating the commands from the editor, the element(s) will try to be placed according to this table: 
+
+|Type | Directory |
+|---|---|
+BATCH | `dataqwik/batch/` |
+DAT | `data/` |
+FKY | `dataqwik/foreign_key/` |
+IDX | `dataqwik/index/` |
+JFD | `dataqwik/journal/` |
+m | `routine/` |
+PROC | `dataqwik/procedure/` |
+properties | `property/` |
+QRY | `dataqwik/query/` |
+RPT | `dataqwik/report/` |
+SCR | `dataqwik/screen/` |
+TABLE | `dataqwik/table/{table_name}/` |
+TRIG | `dataqwik/trigger/` |
+
+If an element is not in the table, a prompt will ask where it should be saved.
+
+In the case of `Table Get from Host`, the TBL and COL files will all be retrieved and placed in the `dataqwik/table/{table_name}/` directory. To get an individual TBL or COL file, use the regular `Get Element from Host` command.
+
+These two commands behave differently when they are used in the Explorer sidebar, specifically on a directory. In this case, both `Get` commands will place the new element(s) inside the targeted directory.
 
-The extension is able to communicate with Host via MRPC121 to do the following:
+### Acting on Tables
 
-* Get/Refresh elements and entire tables from Host
-* Send elements and entire tables to Host
-* Test Compile .PROC and .PSL files
-* Compile and Link .PROC and .PSL files
-* Run .PROC and .PSL files
+When editing a TBL or COL, the commands `Table Refresh from Host` and `Table Send to Host` are present. The `Send` command will act on all existing elements of that table in the project at once, while the `Refresh` command will fetch all elements related to the table from the Host, even elements not present in the project. `Compile and Link` is also available, allowing you to "Rebuild Data Item Control Files".

```

Comment:
Bit nitpicky, but there are some markdown standards, including that lines should not be too long. I think Markdown inherited the 80 column size from other recommendations, which imo is a bit old fashioned. But a few lines here are pretty long. Typescript has a 120 char recommendation, maybe we should follow that in markdown as well?